### PR TITLE
Add validation

### PR DESCRIPTION
**Pull request by gnufied**
- Also rename managmentState key.
- Fix klog initialization when logrus is also present

### CODE REVIEW

File: manifests/4.2.0/local-volumes.crd.yaml [hunk 1]
```diff
@@ -11,3 +11,86 @@ spec:
...
+          type: string
+        kind:
+          description: 'Kind is a string value representing the REST resource this
+            object represents. Servers may infer this from the endpoint the client
+            submits requests to. Cannot be updated. In CamelCase. More info: https://git.k8s.io/community/contributors/devel/api-conventions.md#types-kinds'
+          type: string
+        metadata:
+          type: object
+        spec:
+          properties:
```
Comment (jsafrane) [inline manifests/4.2.0/local-volumes.crd.yaml]: missing managementState and logLevel

File: manifests/4.2.0/local-volumes.crd.yaml [hunk 2]
```diff
@@ -11,3 +11,86 @@ spec:
...
+                required:
+                  - storageClassName
+                  - devicePaths
+                type: object
+              type: array
+          required:
+            - storageClassDevices
+          type: object
+        status:
+          properties:
```
Comment (jsafrane) [inline manifests/4.2.0/local-volumes.crd.yaml]: missing conditions and readyReplicas

**Comment by gnufied** [comment]
@jsafrane fixed. PTAL

**Comment by gnufied** [comment]
Also verified that the operator still works and CRs are created after this change:

```
apiVersion: local.storage.openshift.io/v1
kind: LocalVolume
metadata:
  creationTimestamp: "2019-07-03T18:43:51Z"
  generation: 69
  name: example
  namespace: default
  resourceVersion: "513"
  selfLink: /apis/local.storage.openshift.io/v1/namespaces/default/localvolumes/example
  uid: fbaf930e-09c0-4795-80ae-9737687f30ba
spec:
  logLevel: Normal
  managementState: Managed
  storageClassDevices:
  - devicePaths:
    - /dev/vde
    - /dev/vdf
    fsType: ext4
    storageClassName: foobar
    volumeMode: Filesystem
status:
  conditions:
  - lastTransitionTime: "2019-07-03T18:45:16Z"
    message: Ready
    status: "True"
    type: Available
  generations:
  - group: apps
    hash: ""
    lastGeneration: 1
    name: example-local-provisioner
    namespace: default
    resource: DaemonSet
  - group: apps
    hash: ""
    lastGeneration: 1
    name: example-local-diskmaker
    namespace: default
    resource: DaemonSet
  managementState: Managed
  observedGeneration: 68
  readyReplicas: 0

```

File: pkg/apis/local/v1/types.go [hunk 3]
```diff
@@ -54,9 +54,11 @@ type StorageClassDevice struct {
 	// StorageClass name to use for set of matched devices
 	StorageClassName string `json:"storageClassName"`
 	// Volume mode. Raw or with file system
+	// + optional
 	VolumeMode PersistentVolumeMode `json:"volumeMode"`
```
Comment (jsafrane) [inline pkg/apis/local/v1/types.go]: if it is optional, should it have `omitempty`?

Comment (gnufied) [reply pkg/apis/local/v1/types.go]: fixed

**Comment by gnufied** [comment]
@jsafrane can you PTAL?

**Comment by gnufied** [comment]
I am gonna merge this manually, assuming all concerns were addressed. Anything outstanding we can cover in a follow up PR.

**Pull request merged**